packages/gestalt/src/Sheet.js

```diff
@@ -0,0 +1,278 @@
+// @flow strict
+import React, {
+  forwardRef,
+  useCallback,
+  useState,
+  useEffect,
+  useRef,
+  type Node,
+} from 'react';
+import PropTypes from 'prop-types';
+import classnames from 'classnames';
+import { ESCAPE_KEY_CODE } from './keyCodes.js';
+import { FixedZIndex } from './zIndex.js';
+import Box from './Box.js';
+import Backdrop from './Backdrop.js';
+import focusStyles from './Focus.css';
+import IconButton from './IconButton.js';
+import Heading from './Heading.js';
+import Row from './Row.js';
+import StopScrollBehavior from './behaviors/StopScrollBehavior.js';
+import sheetStyles from './Sheet.css';
+import TrapFocusBehavior from './behaviors/TrapFocusBehavior.js';
+import useReducedMotion from './useReducedMotion.js';
+
+type AnimationType = 'in' | 'out';
+
+type HeadingType = string | Node;
+
+type Props = {|
+  accessibilityDismissButtonLabel: string,
+  accessibilitySheetLabel: string,
+  children?: Node,
+  closeOnOutsideClick?: boolean,
+  footer?: Node,
+  heading?: HeadingType,
+  onDismiss: () => void,
+  size?: 'sm' | 'md' | 'lg' | number,
+|};
+
+const SIZE_WIDTH_MAP = {
+  sm: 540,
+  md: 720,
+  lg: 900,
+};
+
+const DEFAULT_ZINDEX = new FixedZIndex(1);
```

Comment:
@themindfuldev Could we remove the zIndex from the `Sheet` container? If we start with `1`, it'll be very hard to remove moving forward.

I just created #1223 which adds `zIndex` to `<Layer />`. Once that is merged, we can do the following in places where we do need to set zIndex.

```jsx
<Layer zIndex={new FixedZIndex(1)}>
  <Sheet ... />
</Layer>
```

This would also keep it in sync with how `<Modal />` works.

Reply:
Hi @christianvuerings , I tried that yesterday but for some reason that doesn't solve our problems. There's something on `<StopScrollBehavior>` and `<TrapFocusBehavior>` that kind of resets the zIndex and I am forced to set it up again in the element under it. To me it was still being placed under other elements (like the Gestalt Docs header for instance)

Another idea is to pass this `zIndex` as an optional prop so we can allow users to pass a ZIndex if they have to, and default to `FixedZIndex(1)` if that is not provided.

Any thoughts?

Reply:
Example:
<img width="1392" alt="Screen Shot 2020-09-17 at 10 20 44 AM" src="https://user-images.githubusercontent.com/764487/93505272-7ef31500-f8cf-11ea-9dfb-e47d73eabc65.png">

Reply:
@themindfuldev the following does seem to work for me: https://github.com/christianvuerings/gestalt/commit/fb0562fcd2747dbbfdc30ffed01b9bd09c0a5026

Reply:
Thanks @christianvuerings ! Adopted that.